packages/client/src/components/Actions/ActionsList.js

```diff
@@ -1,9 +1,50 @@
-import React from "react";
-import { formatActionString } from "../utils";
+import React, { useContext } from "react";
+import { formatActionString } from "utils";
 import { SIGNER_RESPONSES } from "constants/enums";
+import { useModalContext } from "contexts";
+import { Web3Context } from "contexts/Web3";
+import ActionRequired from "./components/ActionRequired";
+import { useErrorMessage } from "hooks";
 
-function ActionsList({ actions = [], onSign, onConfirm, safeData }) {
+function ActionsList({ actions = [], onSign, onReject, onConfirm, safeData }) {
   const ActionComponents = [];
+
+  const { getActionView } = useContext(Web3Context);
+  const { showErrorModal } = useErrorMessage();
+  const { openModal, closeModal } = useModalContext();
+
+  const onSignAction = async (action) => {
+    try {
+      await onSign(action);
+      closeModal();
+    } catch (error) {
+      showErrorModal(error);
+    }
+  };
+
+  const onRejectAction = async (action) => {
+    try {
+      await onReject(action);
+      closeModal();
+    } catch (error) {
+      showErrorModal(error);
+    }
+  };
+
+  const openSignRejectModal = async (action) => {
```

Comment:
Looks like the modal opened up allows user to sign to approve OR reject, suggest rename to `openApproveOrRejectModal`